packages/angular_devkit/schematics/src/rules/template.ts

```diff
@@ -48,18 +48,22 @@ export interface PathTemplateOptions {
   pipeSeparator?: string;
 }
 
+const decoder = new TextDecoder('utf-8', { fatal: true });
 
 export function applyContentTemplate<T>(options: T): FileOperator {
   return (entry: FileEntry) => {
-    const {path, content} = entry;
-    if (isBinary(content)) {
+    const { path, content } = entry;
+
+    try {
+      const decodedContent = decoder.decode(content);
+
+      return {
+        path,
+        content: Buffer.from(templateImpl(decodedContent, {})(options)),
+      };
+    } catch {
```

Comment:
Can you add a check for the error code `ERR_ENCODING_INVALID_ENCODED_DATA` and re-throw if it is not that code?

Reply:
Done.